internal/serverinstall/ecs.go

```diff
@@ -536,8 +536,7 @@ func (i *ECSInstaller) Upgrade(
 	s = sg.Add("Updating task definition")
 	defer func() { s.Abort() }()
 
-	// assume upgrade to latest
-	if *containerDef.Image == installutil.DefaultServerImage {
+	if *containerDef.Image == installutil.DefaultServerImage && upgradeImg == installutil.DefaultServerImage {
```

Comment:
I know this was already here but you might need a `*containerDef != nil` too to be safe?

Reply:
Updated!